src/test/test-thp.c

```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#include <sys/prctl.h>
+
+#include "tests.h"
+
+#define PR_THP_DISABLE_NOT_SET 0
+#define PR_THP_DISABLE 1
+#define PR_THP_DISABLE_EXCEPT_ADVISED (1 << 1)
+
+static const char *arg_mode = NULL;
+
+static int intro(void) {
+        int r = prctl(PR_GET_THP_DISABLE, 0, 0, 0, 0);
```

Comment:
"int r" is special to us, we always define it on a separate line, because it is *the* place we store return value of all calls we make to. or in other words, as matter of coding style, please do this in two lines, i.e. `int r;` and `r = prctl(…`...

doesn't really matter, but it's a pattern we have followed so far everywhere else.